core/trino-main/src/main/java/io/trino/operator/scalar/ArraysOverlapFunction.java

```diff
@@ -80,11 +81,10 @@ public Boolean arraysOverlapInt(
         int leftPosition = 0;
         int rightPosition = 0;
         while (leftPosition < leftNonNullSize && rightPosition < rightNonNullSize) {
-            long compareValue = comparisonOperator.compare(leftLongArray[leftPosition], rightLongArray[rightPosition]);
```

Comment:
Even if a type is storing its values in a `long`, it's possible that comparison is not a simple less than operation. That's why we need to go through the `comparisonOperator`. This change is probably not correct.
fyi @dain

Reply:
Would love to learn about it if that's the case - and see if there is a way to improve its performance without impacting the correctness. The performance drop using comparisonOperator when the two arrays are large is not insignificant.

Reply:
Maybe try using `TypedSet` instead of sorted arrays to find overlap similar to ArrayIntersectFunction

Reply:
To further improve performance you can byte code generate this for each type.